### PR TITLE
Updated files for issue #2791

### PR DESCRIPTION
**Pull request by Anjali-NEC**
Fix for issue #2791

### CODE REVIEW

File: src/lib/rest/OrionError.cpp [hunk 1]
```diff
@@ -79,7 +79,7 @@ OrionError::OrionError(StatusCode& sc)
 void OrionError::fill(HttpStatusCode _code, const std::string& _details, const std::string& _reasonPhrase)
 {
   code          = _code;
-  reasonPhrase  = _reasonPhrase != ""? _reasonPhrase : httpStatusCodeString(code);
+  reasonPhrase  = !_reasonPhrase.empty()? _reasonPhrase : httpStatusCodeString(code);
```
Comment (fgalan) [inline src/lib/rest/OrionError.cpp:82]:
```suggestion
  reasonPhrase  = _reasonPhrase.empty()? httpStatusCodeString(code) : _reasonPhrase;
```

I think is clerer this way, avoiding negation in the condition and swapping the results.

File: src/lib/rest/OrionError.cpp [hunk 2]
```diff
@@ -92,7 +92,7 @@ void OrionError::fill(HttpStatusCode _code, const std::string& _details, const s
 void OrionError::fill(const StatusCode& sc)
 {
   code          = sc.code;
-  reasonPhrase  = (sc.reasonPhrase != "")? sc.reasonPhrase : httpStatusCodeString(code);
+  reasonPhrase  = (!sc.reasonPhrase.empty())? sc.reasonPhrase : httpStatusCodeString(code);
```
Comment (fgalan) [inline src/lib/rest/OrionError.cpp:95]:
```suggestion
  reasonPhrase  = (sc.reasonPhrase.empty())? httpStatusCodeString(code) : sc.reasonPhrase;
```

**Comment by fgalan** [comment]
Thank you for your contribution!

With regards to code changes, apart from a couple of suggestions I did, I think is fine.

However, note that the associated issue mentions:

> Add that rule in the coding style guide

So please add the corresponding rule to https://github.com/telefonicaid/fiware-orion/blob/master/doc/manuals/contribution_guidelines.md, in the "should" section (it would be rule number S16)

Thanks!

File: src/lib/rest/OrionError.cpp (hunk 1 above)
Comment (Anjali-NEC) [reply src/lib/rest/OrionError.cpp]: Updated

File: src/lib/rest/OrionError.cpp (hunk 2 above)
Comment (Anjali-NEC) [reply src/lib/rest/OrionError.cpp]: Updated

File: doc/manuals/contribution_guidelines.md [hunk 3]
```diff
@@ -750,6 +750,14 @@ void myFunction(const std::string s, std::string* err);
 
 *How to check*: manually
 
+#### S16 (Use empty() for checking zero-length strings)
+
+*Rule*: Use empty() for all checkings with "" and usages of length()
```
Comment (fgalan) [inline doc/manuals/contribution_guidelines.md:755]:
```suggestion
*Rule*: Use empty() for all string empty-ness checking, avoiding comparisons with `""` or `length() == 0`
```

**Comment by Anjali-NEC** [comment]
> So please add the corresponding rule to https://github.com/telefonicaid/fiware-orion/blob/master/doc/manuals/contribution_guidelines.md, in the "should" section (it would be rule number S16)

Added rule in https://github.com/telefonicaid/fiware-orion/blob/master/doc/manuals/contribution_guidelines.md.

File: doc/manuals/contribution_guidelines.md [hunk 4]
```diff
@@ -750,6 +750,14 @@ void myFunction(const std::string s, std::string* err);
...
+
+*Rationale*: empty() is the best way of checking for zero-length strings.
```
Comment (fgalan) [inline doc/manuals/contribution_guidelines.md:757]:
```suggestion
*Rationale*: empty() is [the best way of checking for zero-length strings](http://stackoverflow.com/questions/41425569/checking-for-empty-string-in-c-alternatives).
```

To include a link to the place were the discussion took place.

File: doc/manuals/contribution_guidelines.md (hunk 3 above)
Comment (Anjali-NEC) [reply doc/manuals/contribution_guidelines.md]: Updated

File: doc/manuals/contribution_guidelines.md (hunk 4 above)
Comment (Anjali-NEC) [reply doc/manuals/contribution_guidelines.md]: Updated

**Review by fgalan — APPROVED** [review]
LGTM. Thanks!

**Comment by Chandradeep-NEC** [comment]
@fgalan 
PR is still open but the issue has been closed. As you have marked LGTM for this PR, request to merge the PR.

**Comment by fisuda** [comment]
I have just sent the PR #3690. It would be great if you could review the PR.

**Pull request merged**